server/application/application.go

```diff
@@ -980,6 +981,14 @@ func (s *Server) getAppLiveResource(ctx context.Context, action string, q *appli
 		return nil, nil, nil, err
 	}
 
+	config, err := s.getApplicationClusterConfig(ctx, a)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("error getting application cluster config: %w", err)
+	}
+	if q.GetKind() == "Application" && q.GetGroup() == "argoproj.io" && *q.Name == q.GetResourceName() {
+		return nil, config, a, nil
+	}
+
```

Comment:
I think this would break the RBAC model, because `resources, update` would now imply `applications, update`.

Instead of modifying `getAppLiveResource`, could you update `ListResourceActions` to call this directly?

Reply:
Yeah, fair point, I'll change that.

I've been trying to work out how to test this, but I can't find any existing unit tests for this functionality, and it would require mocking the server cache, whose first mockable Interface is about three levels of cache down the composition stack...

Do you mind if I continue to leave this not covered by unit tests?

Reply:
I think it'll be okay without tests... I'll review again Friday (I hope? This week has had a lot of surprises...) and decide then. Might take a crack at adding some, if just for the learning experiience.